pkg/services/store/object/object.proto

```diff
@@ -0,0 +1,254 @@
+syntax = "proto3";
+package object;
+
+option go_package = "./;object";
+
+// Will be replaced with something from the SDK
+message UserInfo {
+  // internal grafana user ID
+  int64 id = 1;
+
+  // login name
+  string login = 2; // string ID? 
+}
+
+// The canonical object/document data -- this represents the raw bytes and storage level metadata
+message RawObject {
+  // Unique ID
+  string UID = 1;
+
+  // Identify the object kind.  This kind will be used to apply a schema to the body and
+  // will trigger additional indexing behavior.
+  string kind = 2;
+
+  // Time in epoch milliseconds that the object was created
+  int64 created = 3;
+
+  // Time in epoch milliseconds that the object was modified
+  int64 modified = 4;
+
+  // Who created the object
+  UserInfo created_by = 5;
+
+  // Who modified the object
+  UserInfo modified_by = 6;
+
+  // Content Length
+  int64 size = 7;
+
+  // MD5 digest of the body
+  string ETag = 8;
+
+  // Raw bytes of the storage object.  The kind will determine what is a valid payload
+  bytes body = 9;
+
+  // The version will change when the object is saved. It is not necessarily sortable
+  //
+  // NOTE: currently managed by the dashboard+dashboard_version tables
+  string version = 10;
+
+  // optional "save" or "commit" message
+  //
+  // NOTE: currently managed by the dashboard_version table, and will be returned from a "history" command
+  string comment = 11;
+
+  // Location (path/repo/etc) that defines the canonocal form
+  //
+  // NOTE: currently managed by the dashboard_provisioning table
+  string sync_src = 12;
+
+  // Time in epoch milliseconds that the object was last synced with an external system (provisioning/git)
+  //
+  // NOTE: currently managed by the dashboard_provisioning table
+  int64 sync_time = 13;
+}
+
+// Report error while working with objects
+// NOTE: real systems at scale will contain errors.
+message ObjectErrorInfo {
+  // Match an error code registry?
+  int64 code = 1;
+
+  // Simple error display
+  string message = 2;
+
+  // Details encoded in JSON
+  bytes details_json = 3;
+}
+
+//-----------------------------------------------
+// Get request/response
+//-----------------------------------------------
+
+message ReadObjectRequest {
+  string UID = 1;
```

Comment:
for the UID+kind... - should we at least comment the expected behavior for read/delete/history? e.g. one of those:

- you need to pass both `UID` and `kind` - both need to match 
- `kind` is optional. if its passed, it needs to match stored object's kind

Reply:
I added comments... I am also 99% sure this should be replaced with a single "GRN" concept that holds the two together